sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -0,0 +1,243 @@
+package com.mesosphere.sdk.offer;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.Resource.DiskInfo;
+import org.apache.mesos.Protos.Value;
+import org.apache.mesos.Protos.Volume;
+
+import com.mesosphere.sdk.specification.ResourceSpec;
+
+/**
+ * Constructs Mesos {@link Resource} protobufs.
+ */
+public class ResourceBuilder {
+
+    /** The name used for reserved network port resources. */
+    public static final String PORTS_RESOURCE_TYPE = "ports";
+    /** The name used for reserved storage/disk resources. */
+    public static final String DISK_RESOURCE_TYPE = "disk";
```

Comment:
I think that'll happen as we start moving away from passing around mesos protobufs internally